activerecord/lib/active_record/reflection.rb

```diff
@@ -697,7 +706,12 @@ def through_reflection
       def chain
         @chain ||= begin
           a = source_reflection.chain
-          b = through_reflection.chain
+          b = through_reflection.chain.map(&:dup)
```

Comment:
I thought it had something to do with the caching but this was changed a long time ago and removing it no longer breaks. We must have changed something else that this is no longer needed. I'll remove it. :smile: